R/skeleton.R

```diff
@@ -56,19 +58,25 @@ book_skeleton = function(
   for (i in seq_along(rmd_files)) {
     content = c(titles[i], '')
     if (i == 1) {
+      default_metadata <- list(
+        title = title,
+        author = author,
+        documentclass = documentclass
+      )
+      index_metadata = modifyList(default_metadata, as.list(index_metadata))
       content = c(
-        '---', sprintf('title: "%s"', title), sprintf('author: "%s"', author),
-        sprintf('documentclass: "%s"', documentclass),
-        'site: bookdown::bookdown_site', '---', '', content,
+        '---', yaml::as.yaml(index_metadata), '---', '',
+        content,
         'Start writing your book here. If you are in RStudio,',
         'Click the Build button to build the book.'
       )
     }
-    write_file(content, rmd_files[i])
+    write_file(content, file.path(path, rmd_files[i]))
   }
   write_file(
-    sprintf('bookdown::%s: default', c('gitbook', 'pdf_book', 'epub_book')), '_output.yml'
+    sprintf('bookdown::%s: default', c('gitbook', 'pdf_book', 'epub_book')),
+    file.path(path, '_output.yml')
   )
-  write_file(sprintf('book_filename: %s', name), '_bookdown.yml')
+  write_file(sprintf('book_filename: %s', name), file.path(path, '_bookdown.yml'))
```

Comment:
why change the directory rather than using `file.path()`?

Reply:
I find it just help avoid repetition (every writing needs to `file.path(path, ...)`, it must not be forgotten) and DNRY is something good., Also it could make it easier to read 

* `xfun::in_dir("my_dir",  something_in_another_dir())`
* `withr::with_dir("my_dir", something_in_another_dir())`

But that is all. No benefit in the R side of things. It work the same. 

Let's leave it like that